VirtoCommerce.Storefront/Controllers/Api/ApiOrderController.cs

```diff
@@ -180,6 +180,28 @@ public async Task<ActionResult> GetInvoicePdf(string orderNumber)
             return File(stream, "application/pdf");
         }
 
+        // PUT: storefrontapi/orders/{orderNumber}/status
+        [HttpPut("{orderNumber}/status")]
+        [ValidateAntiForgeryToken]
+        public async Task<ActionResult> ChangeOrderStatus(string orderNumber, [FromBody] dynamic payload)
+        {
+            var newStatus = payload.newStatus.ToString();
+
+            if (string.IsNullOrEmpty(newStatus))
+            {
+                throw new StorefrontException("The new order status can't be nullable or empty");
+            }
+
+            using (await AsyncLock.GetLockByKey(GetAsyncLockKey(orderNumber, WorkContext)).LockAsync())
+            {
+                var orderDto = await GetOrderDtoByNumber(orderNumber);
+                orderDto.Status = newStatus;
```

Comment:
Any user can set any status?

Reply:
You must add authorization for this API, it will be the 100% vulnerability if any user with any privileges (even anonymous) can change the status of any order in the system simply by calling this API.

Reply:
`GetOrderDtoByNumber()` checks if current user can access the order, but even if this is true it doesn't mean current user can modify this order.

Reply:
The idea is that user management layer (for procurement portal) will be built on the client-side and we going to permit order status changing only for platform users with appropriate rights (must be admin). In this case, the admin credential will be as a proxy. How we should verify our permissions in the right way to handle this case?